### PR TITLE
WIP: Remote SR

### LINKED ISSUE #291: Activate Google cloud SR from backend
As a user/customer I want that my commands are recognized by Google speech recognition

40 SP

Tasks:
- [x] wakeup works with sphinx
- [x] Find a way to activate Google SR from backend (overhead as little as possible)
- [x] Listen to speech input from user and send result back to device
- [x] Send result to parser

### PR DESCRIPTION
**Pull request by Bensge**
- [x] Send data back to Server via HTTP-Rest request
- [x] Start Chrome automatically ~(maybe)~
- [x] Test Chrome launching on Windows 
-> Remaining issue: User profile loading doesn't work (file permission error)
- [x] Test Chrome launching on Linux
- [ ] Make sure it behaves correctly in Docker
- [ ] Integrate into local speech system in core project
-> @MakinZeel will do this task: #322 
- [x] Test with Raspberry Pi
-> Does not work and never will (no Chrome for ARM linux)

Once finished, close #291

### CODE REVIEW

File: http-server/pom.xml [hunk 1]
```diff
@@ -44,6 +44,11 @@
 			<artifactId>jersey-hk2</artifactId>
 			<version>${jersey.version}</version>
 		</dependency>
+		<dependency>
+			<groupId>org.glassfish.jersey.media</groupId>
+			<artifactId>jersey-media-sse</artifactId>
+			<version>LATEST</version>
```
Comment (Legion2) [inline http-server/pom.xml]: don't use `LATEST`

File: remote-sr/pom.xml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		<dependency>
+			<groupId>de.unistuttgart.iaas.amyassist</groupId>
+			<artifactId>amy-api</artifactId>
+			<scope>provided</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.slf4j</groupId>
+			<artifactId>slf4j-api</artifactId>
+			<version>${slf4j.version}</version>
+			<scope>provided</scope>
```
Comment (Legion2) [inline remote-sr/pom.xml]: remove the `<scope>provided</scope>` everywhere

File: remote-sr/src/main/resources/index.html [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	</body>
+	<script>
+		/**
+		 * Simple Speech Recognition website.
+		 * @author Benno Krauß
+		 */
+		if (!('webkitSpeechRecognition' in window)) {
+			console.log('Browser doesn\'t support SR!');
+		}
+		var recognition = new webkitSpeechRecognition();
```
Comment (Legion2) [inline remote-sr/src/main/resources/index.html]: use `const` instead of `var`

File: remote-sr/src/main/java/de/unistuttgart/iaas/amyassist/amy/remotesr/SSEResource.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+    @Reference
+    private RemoteSR sr;
+
+    @GET
+    @Path("eventstream")
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    public void eventStream(@Context SseEventSink sink, @Context Sse sse) {
+
+        sr.setResource(this);
```
Comment (Legion2) [inline remote-sr/src/main/java/de/unistuttgart/iaas/amyassist/amy/remotesr/SSEResource.java]: don't do this

File: amy-master-node/pom.xml [hunk 5]
```diff
@@ -45,6 +45,16 @@
 			<artifactId>amy-message-hub</artifactId>
 			<version>${amy.message-hub.version}</version>
 		</dependency>
+		<dependency>
+			<groupId>de.unistuttgart.iaas.amyassist</groupId>
+			<artifactId>amy-registry</artifactId>
```
Comment (Legion2) [inline amy-master-node/pom.xml]: why?

File: remote-sr/src/main/java/de/unistuttgart/iaas/amyassist/amy/remotesr/StaticWebsite.java [hunk 6]
```diff
@@ -52,11 +53,14 @@
...
     public String getStaticWebsite() {
         try {
-            URI filePath = getClass().getClassLoader().getResource("index.html").toURI();
-            return new String(Files.readAllBytes(Paths.get(filePath)), StandardCharsets.UTF_8);
-        } catch (URISyntaxException | IOException | NullPointerException e) {
+            URL file = getClass().getClassLoader().getResource("index.html");
+            if (file != null) {
+                URI filePath = file.toURI();
+                return new String(Files.readAllBytes(Paths.get(filePath)), StandardCharsets.UTF_8);
+            }
```
Comment (Legion2) [inline remote-sr/src/main/java/de/unistuttgart/iaas/amyassist/amy/remotesr/StaticWebsite.java]: throw a exception if the static resource is not found

File: remote-sr/pom.xml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+			<artifactId>amy-utility</artifactId>
+			<version>${amy.utility.version}</version>
+		</dependency>
+		<dependency>
+			<groupId>javax.ws.rs</groupId>
+			<artifactId>javax.ws.rs-api</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>com.eclipsesource.minimal-json</groupId>
+			<artifactId>minimal-json</artifactId>
```
Comment (Legion2) [inline remote-sr/pom.xml]: why is this needed?

Comment (Bensge) [reply remote-sr/pom.xml]: not needed, I'll remove it

**Comment by Legion2** [comment]
CHROME_PATH should be a config property

File: remote-sr/src/main/java/de/unistuttgart/iaas/amyassist/amy/remotesr/RemoteSR.java [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+    /**
+     * launch the chrome browser, open the remote-SR url immediately and prevent
+     * microphone permission dialog by using a prepared user profile
+     */
+    private void launchChrome() {
+        try {
+            URL fileURL = getClass().getClassLoader().getResource("chrome_profile");
+            if (fileURL == null) {
+                throw new IOException("Couldn't get chrome_profile resource");
```
Comment (Legion2) [inline remote-sr/src/main/java/de/unistuttgart/iaas/amyassist/amy/remotesr/RemoteSR.java]: throw a RuntimeException and don't catch it, so the caller of `launchChrome` knows that something failed

File: remote-sr/src/main/java/de/unistuttgart/iaas/amyassist/amy/remotesr/RemoteSR.java [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+            }
+            String file = fileURL.getFile();
+
+            String chromePath = configurationManager.getConfiguration(CONFIG_NAME)
```
Comment (Legion2) [inline remote-sr/src/main/java/de/unistuttgart/iaas/amyassist/amy/remotesr/RemoteSR.java]: use a default config

**Comment by Legion2** [comment]
#322
